subworkflows/local/input_check.nf

```diff
@@ -8,53 +8,64 @@ include { SAMPLESHEET_CHECK } from '../../modules/local/samplesheet_check' addPa
 
 workflow INPUT_CHECK {
     take:
-    samplesheet // file: /path/to/samplesheet.csv
+        samplesheet // file: /path/to/samplesheet.csv
 
     main:
-    SAMPLESHEET_CHECK ( samplesheet )
-        .splitCsv ( header:true, sep:',' )
-        .set { sheet }
+        SAMPLESHEET_CHECK ( samplesheet )
+            .splitCsv ( header:true, sep:',' )
+            .set { sheet }
 
-    reads = sheet.map { create_fastq_channels(it) }
-    sample = sheet.map { create_sample_channels(it) }
+        ch_case_info = sheet.first()
+                            .map { create_case_channel(it) }
+        reads        = sheet.map { create_fastq_channel(it) }
+        samples      = sheet.map { create_samples_channel(it) }
 
     emit:
-    reads // channel: [ val(meta), [ reads ] ]
-    sample // channel: [ sample, sex, phenotype, paternal_id, maternal_id, case_id ]
+        ch_case_info    // channel: [ case_id ]
+        reads           // channel: [ val(meta), [ reads ] ]
+        samples         // channel: [ sample_id, sex, phenotype, paternal_id, maternal_id, case_id ]
 }
 
 // Function to get list of [ meta, [ fastq_1, fastq_2 ] ]
-def create_fastq_channels(LinkedHashMap row) {
+def create_fastq_channel(LinkedHashMap row) {
+    // create meta map
     def meta = [:]
     meta.id           = row.sample
     meta.single_end   = row.single_end.toBoolean()
 
-    // TODO: add read group to the meta map
-
-    def array = []
+    // add path(s) of the fastq file(s) to the meta map
+    def fastq_file_meta = []
     if (!file(row.fastq_1).exists()) {
         exit 1, "ERROR: Please check input samplesheet -> Read 1 FastQ file does not exist!\n${row.fastq_1}"
     }
     if (meta.single_end) {
-        array = [ meta, [ file(row.fastq_1) ] ]
+        fastq_file_meta = [ meta, [ file(row.fastq_1) ] ]
     } else {
         if (!file(row.fastq_2).exists()) {
             exit 1, "ERROR: Please check input samplesheet -> Read 2 FastQ file does not exist!\n${row.fastq_2}"
         }
-        array = [ meta, [ file(row.fastq_1), file(row.fastq_2) ] ]
+        fastq_file_meta = [ meta, [ file(row.fastq_1), file(row.fastq_2) ] ]
     }
-    return array
+    return fastq_file_meta
 }
```

Comment:
@ramprasadn This part comes shipped with the pipeline when creating a template via `nf-core create`. It's better to make a PR here: https://github.com/nf-core/tools/blob/master/nf_core/pipeline-template/subworkflows/local/input_check.nf and discuss the change with the core members + community. Another reason why changing it here is bad b/c it'll become out of sync with future template updates.

Reply:
Good point! I didn't think it'd matter because this script is under local subdirectory, but If that's not the case then its a bit misleading and it should be changed. Will discuss this with Harshil :)